tools/mtrace/mtrace-reader.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: BSD-3-Clause
+#
+# Copyright (c) 2022, Intel Corporation. All rights reserved.
+
+#pylint:disable=mixed-indentation
+
+# Tool to stream data from Linux SOF driver "mtrace" debugfs
+# interface to standard output. Plain "cat" is not sufficient
+# as each read() syscall returns log data with a 32bit binary
+# header, containing the payload length.
+
+import struct
+import os
+import sys
+
+READ_BUFFER = 16384
+MTRACE_FILE = "/sys/kernel/debug/sof/mtrace/core0"
+
+fd = os.open(MTRACE_FILE, os.O_RDONLY)
+while fd >= 0:
```

Comment:
I have no idea about .py. How does this `os.open()`, `os.read()` compare to `with open(MTRACE_FILE) as file:`?

Reply:
@lyakh  This is a feature of the debugfs interface. Every read system call returns as much data as the application requests for. The first 32bit tells how much of that data is actual log data. Due to this protocol, all higher level file reading abstractions will not be able to handle the debugfs interface. Instead we use direct os.read() calls without any intermediate buffering.